perceval/backend.py

```diff
@@ -510,6 +533,206 @@ def setup_cmd_parser(cls):
+                                   archived_after)
+
+        for filepath in filepaths:
+            self.backend.archive = Archive(filepath)
+            items = self.backend.fetch_from_archive()
+
+            try:
+                for item in items:
+                    yield item
+            except ArchiveError as e:
+                logger.warning("Ignoring %s archive due to: %s", filepath, str(e))
+
+
+class Summary:
+    """Summary class for fetch executions.
+
+    This class models the summary of a fetch execution. It includes
+    the last UUID, number of items fetched, skipped and their sum,
+    plus the min, max and last updated_on times. Furthermore, for
+    backends using offsets, the corresponding summary contains the
+    min and max offsets retrieved. Finally, the summary also includes
+    some extra fields, which can be used by any backend to include
+    fetch-specific information.
+    """
+    def __init__(self):
+        self.fetched = 0
+        self.skipped = 0
+        self.min_updated_on = None
+        self.max_updated_on = None
+        self.last_updated_on = None
+        self.last_uuid = None
+        self.min_offset = None
+        self.max_offset = None
+        self.extras = None
+
+    @property
+    def total(self):
+        return self.fetched + self.skipped
+
+    def update(self, item):
+        """Update the summary attributes by accessing the item data
+
+        :param item: a Perceval item
+        """
+        self.fetched += 1
+        self.last_uuid = item['uuid']
+
+        updated_on = unixtime_to_datetime(item['updated_on'])
+        self.min_updated_on = updated_on if not self.min_updated_on else min(self.min_updated_on, updated_on)
+        self.max_updated_on = updated_on if not self.max_updated_on else max(self.max_updated_on, updated_on)
+        self.last_updated_on = updated_on
+
+        offset = item.get('offset', None)
+        if offset is not None:
+            self.min_offset = offset if self.min_offset is None else min(self.min_offset, offset)
+            self.max_offset = offset if self.max_offset is None else max(self.max_offset, offset)
+
+    def to_dict(self):
+        """Convert the summary object to a dict"""
+
+        summary = {
+            'summary': {
+                'total': self.total,
+                'fetched': self.fetched,
+                'skipped': self.skipped,
+                'min_updated_on': self.min_updated_on.isoformat() if self.min_updated_on else None,
+                'max_updated_on': self.max_updated_on.isoformat() if self.max_updated_on else None,
+                'last_updated_on': self.last_updated_on.isoformat() if self.last_updated_on else None,
+                'last_uuid': self.last_uuid,
+                'min_offset': self.min_offset,
+                'max_offset': self.max_offset,
+                'extras': self.extras
+            }
+        }
+
+        return summary
+
+    def __repr__(self):
```

Comment:
I don't really like to have this here. `__repr__` should write something less complex. I don't think we need it on this case.